src/main/java/io/appform/dropwizard/sharding/execution/TransactionExecutor.java

```diff
@@ -47,70 +47,29 @@ public TransactionExecutor(final ShardInfoProvider shardInfoProvider,
         this.observer = observer;
     }
 
-    public <T, U> Optional<T> executeAndResolve(SessionFactory sessionFactory, boolean readOnly, Function<U, T> function, U arg,
-                                                String opType,
-                                                int shardId) {
-        T result = execute(sessionFactory, readOnly, function, arg, opType, shardId);
-        return Optional.ofNullable(result);
-    }
-
-    public <T, U> T execute(SessionFactory sessionFactory, boolean readOnly, Function<U, T> function, U arg,
-                            String opType,
-                            int shardId) {
-        return execute(sessionFactory, readOnly, function, arg, t -> t, opType, shardId);
-    }
-
-    public <T, U> T execute(SessionFactory sessionFactory, boolean readOnly, Function<U, T> function, U arg, boolean completeTransaction,
-                            String opType,
-                            int shardId) {
-        return execute(sessionFactory, readOnly, function, arg, t -> t, completeTransaction, opType, shardId);
-    }
-
-    public <T, U, V> V execute(SessionFactory sessionFactory, boolean readOnly, Function<U, T> function, U arg, Function<T, V> handler,
-                               String opType,
-                               int shardId) {
-        return execute(sessionFactory, readOnly, function, arg, handler, true, opType, shardId);
-    }
-
-    public <T, U, V> V execute(SessionFactory sessionFactory,
-                               boolean readOnly,
-                               Function<U, T> function,
-                               U arg,
-                               Function<T, V> handler,
-                               boolean completeTransaction,
-                               String opType,
-                               int shardId) {
-        return execute(sessionFactory,
-                readOnly,
-                session -> {
-                    T result = function.apply(arg);
-                    return handler.apply(result);
-                },
-                completeTransaction,
-                opType,
-                shardId);
+    public <T> T execute(SessionFactory sessionFactory, boolean readOnly, String commandName,
+        OpContext<T> opContext,
```

Comment:
fix the indentation, have each param in a separate line

Reply:
Done.